server/index.ts

```diff
@@ -122,6 +136,35 @@ async function start(id: number, disconnect: () => void) {
       }`
     );
   });
+  server.on("shutdown", () => {
+    // This is done so that new rate limiter
+    // configurations would immediately come into effect
+    // on server restart. As an example,
+    // the `duration` property doesn't change
+    // for already created keys, but affects all new keys.
+    //
+    // Refer https://github.com/animir/node-rate-limiter-flexible/wiki/Options#clearexpiredbytimeout
+    Logger.info("lifecycle", `Deleting all rate limiter specific keys...`);
+
+    const scanStream = Redis.defaultClient.scanStream({
+      match: "rlflx:*",
+      count: 1000,
+    });
+
+    scanStream.on("data", (keys) => {
+      if (!isEmpty(keys)) {
+        Redis.defaultClient.unlink(keys, (err) => {
+          if (err) {
+            Logger.error("lifecycle", err);
+          }
+        });
+      }
+    });
+
+    scanStream.on("end", () => {
```

Comment:
Does server shutdown need to wait on this completing…

Reply:
Not necessarily. As long as the existing keys are flushed, we should be good. We can remove this.